### ISSUE TITLE
Runtime Error loading undefined.js after including KnockoutJS

### ISSUE THREAD

**Issue by theoutlander** [nikhilk/scriptsharp#314]
I added a reference to KO and tried to run the project (without referencing KO JS in HTML), but I'm running into a two issues.
- **The generated Javascript has the following statement:**
  
  define('DomeClient', ['ss', 'jquery', 'knockout'], function(ss, $, ko) {
  
  This causes ssloader to load scripts/knockout.js (which doesn't exist) instead of scripts/knockout-2.1.0.js. Do I need to do anything special so that it looks for the correct file? A module definition somewhere?
- **I manually modified the file so that it looks for knockout-2.1.0.js, but when it loads that file there's another weird error that I see:**
  
  GET http://localhost:58318/scripts/undefined.js 404 (Not Found) ssloader.js:229
  _loadScript ssloader.js:229
  _loadModule ssloader.js:284
  (anonymous function) ssloader.js:341
  each ssloader.js:15 (tems[i] is undefined here)
  require ssloader.js:340
  define ssloader.js:355
  ko knockout-2.1.0.debug.js:15
  (anonymous function) knockout-2.1.0.debug.js:20
  (anonymous function)

**Comment by nikhilk** [comment]
Did you checkout the Knockout Hello World sample - see https://github.com/nikhilk/scriptsharp/tree/cc/samples/KOWorld.

Things to look for in that sample:
- Knockout script is added to web app via a nuget package.
- Sample page uses require.js script loader (more on that below)
- Script# project defines the generic "knockout" script assembly name to a version-specific module in AssemblyInfo.cs using the following attribute:
  
  [assembly: ScriptReference("knockout", Path = "knockout-2.1.0")]

On the 2nd bullet point, the script# loader is more optimized/smaller while supporting some other optimization scenarios, but does so at the cost of not supporting anonymous modules (i.e. modules that call define without naming the module). Supposedly not supporting anonymous modules is one of the valid subsets of the AMD pattern. Unfortunately (IMO) the knockout module is one example of a module that defines its module without specifying a name... hence it doesn't work with the script# loader.

Of course one alternative is to have a fork of knockout that does add a module name when defining the module... and I've toyed with the idea of creating that, but chose to put that off, since I didn't want to get in the business of managing a knockout fork. Besides, there is a well-established script loader that does handle the scenario, so might as well show script# works with that bit of the ecosystem.

Hopefully this gets your scenario working...

**Comment by theoutlander** [comment]
Thanks, making the change in AssemblyInfo worked. I didn't have to do that in 0.7 and this was the first time I tried KO in 0.8. I had taken a look at the sample project earlier today, but missed that detail as I started to panic (got a dry-run in an hour for a symposium and unfortunately I'm running into some last minute S# issues).

[On that note, I had emailed you 2-3 weeks ago for collaborating on the documentation but didn't hear back (wonder if you got my email? Sent from theoutlander at live).]

I had to change some things around and was able to get it to work with require.js. Would be nice if I could document these kind of things once we touch base.
